tasmota/tasmota_xsns_sensor/xsns_02_analog.ino

```diff
@@ -707,6 +710,48 @@ void AdcShow(bool json) {
         }
         break;
       }
+      case ADC_WIND_DIR: {
+        uint16_t analog = AdcRead(Adc[idx].pin, 5);
+        float dir_val = analog * ANALOG_V33 / ANALOG_RANGE;
+        char dir_chr[FLOATSZ];
+        dtostrfd(dir_val, 2, dir_chr);
+        String wind_chr = D_ERROR;
+
+        if(dir_val > 2.50 && dir_val < 2.70 ) {
+          wind_chr = D_TX20_NORTH;
+        }
+        if(dir_val > 1.30 && dir_val < 1.70 ) {
+          wind_chr = D_TX20_NORTH D_TX20_EAST;
+        }
+        if(dir_val > 0.20 && dir_val < 0.40 ) {
+          wind_chr = D_TX20_EAST;
+        }
+        if(dir_val > 0.50 && dir_val < 0.74 ) {
+          wind_chr = D_TX20_SOUTH D_TX20_EAST;
+        }
+        if(dir_val > 0.75 && dir_val < 1.20 ) {
+          wind_chr = D_TX20_SOUTH;
+        }
+        if(dir_val > 2.00 && dir_val < 2.20 ) {
+          wind_chr = D_TX20_SOUTH D_TX20_WEST;
+        }
+        if(dir_val > 3.00 && dir_val < 3.25 ) {
+          wind_chr = D_TX20_WEST;
+        }
+        if(dir_val > 2.8 && dir_val < 2.99 ) {
+          wind_chr = D_TX20_NORTH D_TX20_WEST;
+        }
```

Comment:
Number of tests can be halved in 2 by ordering properly and using else if
Also this leaves empty holes like 2.995 which falls in no cases

Just an opinion: I know you didn't invented those direction macros but wouldn't an angle easier to process downstream in a JSON data ?
I could make sense to use human readable directions in hte Web UI but angle in the JSON

Reply:
I agree that an angle would be preferable for JSON, and the web UI is not meant to be "the useful place" anyway.

And it does look odd that values 
0.75..1.20 is south
1.30..1.70 is northeast
2.00..2.20 is southwest
2.50..2.70 is north
I was expecting the analog value to have a circular dependence on the angle, but your values are jumping around. Is that really correct?

Reply:
It uses a resistor network with reed switches according to the data sheet, https://cdn.sparkfun.com/assets/d/1/e/0/6/DS-15901-Weather_Meter.pdf

Reply:
I suggest to convert the received values to degrees and use that to index in the winddirection text string like TX20 does but then without the NNE options. Then output both string and degrees again like TX20 does.

Reply:
In addition move from ADC float ASAP to degree integer lowering code size considerably.